server/middleware/populateArchivedLocations.ts

```diff
@@ -0,0 +1,34 @@
+import { RequestHandler } from 'express'
+import { Services } from '../services'
+import decorateLocation from '../decorators/location'
+import logger from '../../logger'
+
+export default function populateArchivedLocations({
+  authService,
+  locationsService,
+  manageUsersService,
+}: Services): RequestHandler {
+  return async (req, res, next) => {
+    const { user, prisonId } = res.locals
+
+    try {
+      const token = await authService.getSystemClientToken(user.username)
+      res.locals.archivedLocations = await Promise.all(
```

Comment:
will this work with just an 'await' rather than promise.all as there is only 1 async call in there?

Reply:
This Promise.all is awaiting the result of the `.map(location =>` just further down